framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -564,35 +579,33 @@ public int runTests() throws Exception {
     return totalExecutionFailures + totalDataVerificationFailures + totalPlanVerificationFailures + totalTimeoutFailures + totalRandomFailures;
   }
 
-  public void setup() throws IOException, InterruptedException {
-    if (!new File(drillOutputDir).exists()) {
-      new File(drillOutputDir).mkdir();
+  public void setup() throws IOException, InterruptedException, URISyntaxException {
+    if (!new File(DrillTestDefaults.DRILL_OUTPUT_DIR).exists()) {
+      new File(DrillTestDefaults.DRILL_OUTPUT_DIR).mkdir();
     }
 
     LOG.info("> Uploading storage plugins\n");
-    String templatePath = CWD + "/conf/plugin-templates/";
-    LOG.info(">> Path: " + templatePath + "\n");
-    File[] templateFiles = new File(templatePath).listFiles();
-    for (File templateFile : templateFiles) {
-      String filename = templateFile.getName();
-      LOG.info(">> File: " + filename);
-      String pluginType = filename.substring(0, filename.indexOf('-'));
-      Utils.updateDrillStoragePlugin(templateFile.getAbsolutePath(),
-    		drillStoragePluginServer, pluginType, fsMode);
-      Thread.sleep(200);
+    String templatePath = DrillTestDefaults.CWD + "/conf/plugin-templates/";
+    Utils.updateDrillStoragePlugins(templatePath);
+
+    if (DrillTestDefaults.IS_SECURE_CLUSTER) {
+      templatePath = DrillTestDefaults.CWD + "/conf/hive-plugin-templates-secure/";
```

Comment:
Can this be made a generic secure dir, rather than specific to hive?

Reply:
Also, just to understand better, we first upload all templates in the original dir. And then depending on it's secure or unsecure, we additionally upload the templates in one of the dirs? Perhaps we could make it more generic & structured? Say separating the contents in the original dir into "common", "secure" & "unsecure". Common should contain all plugins which don't have to be modified depending on security. Secure should contain all templates which are to be secure. And same with unsecure.

Reply:
Yes, I am using the second approach. 

The only templates which are affected by the cluster being secure or not is hive. So, I now have two sets of hive templates. First the common ones are uploaded and then either secure hive or unsecure hive

Reply:
I was suggesting we create those sub folders with all templates in one among them. And make this generic, rather than specific to hive.

Reply:
done